cime/scripts/tests/scripts_regression_tests.py

```diff
@@ -152,6 +152,17 @@ def test_lib_doctests(self):
                         run_cmd_assert_result(self, 'PYTHONPATH={}:$PYTHONPATH python -m doctest {} 2>&1'.format(LIB_DIR, filepath), from_dir=LIB_DIR)
                     else:
                         print("{} has no doctests".format(filepath))
+###############################################################################
+class C_TestGridGeneration(unittest.TestCase):
+###############################################################################
+    def test_gen_domain(self):
+        inputdata = MACHINE.get_value("DIN_LOC_ROOT")
```

Comment:
Oh, almost forgot; You should skip this test if the model != e3sm. We don't want to make CESM/UCF folks wait for a test of a tool they don't care about.